online_meetings/canvas_zoom_meetings.py

```diff
@@ -97,26 +100,36 @@ def zoom_course_report(canvas_account=1, enrollment_term_id=1, published=True):
                     zoom_session.headers.update({
                         'X-XSRF-TOKEN': pattern.group(1)
                     })
-                # Get tab 1 (Previous Meetings)
-                    data = {'page': 1,
-                            'total': 0,
-                            'storage_timezone': 'America/Montreal',
-                            'client_timezone': 'America/Detroit'}
-                    r = zoom_session.get("https://applications.zoom.us/api/v1/lti/rich/meeting/history/COURSE/all", params=data)
-                    zoom_json = json.loads(r.text)
-
-                    for meeting in zoom_json["result"]["list"]:
-                        zoom_courses_meetings.append({
-                            'course_id': course.id,
-                            'meeting_id': meeting['meetingId'],
-                            'meeting_number': meeting['meetingNumber'],
-                            'host_id': meeting['hostId'],
-                            'topic': meeting['topic'],
-                            'join_url': meeting['joinUrl'],
-                            'start_time': meeting['startTime'],
-                            'status': meeting['status'],
-                            'timezone': meeting['timezone']
-                        })
+
+                    # Page iterator
+                    page_num = 1
+                    while True:
```

Comment:
Nitpick: don't love `while True ... break`.

Reply:
Yeah, I don't either. I was thinking about refactoring this but it would be a lot more code. This API call isn't great and you have to make one call first. Then it tells you the number of items per page and the total items. You could set up a for loop based on that information or just break when it runs out of pages to return. 

I don't like while True because you might get into an endless loop. I'm also using a 
`while response.status_code == requests.codes.too_many_requests:` which is essentially a potential endless loop too if it's always returning too_many_requests. This shouldn't be as likely though. I think I left this in from the quick hack but maybe this can be refactored to be better.

Reply:
So far, it looks like these loops are okay. I don't think it would take too much work to do a simple refactor where you define a variable holding a boolean before. Like what I'm doing in `inventory.py`:

https://github.com/tl-its-umich-edu/course-inventory/blob/373ea575e5ea6830f68a022a02cd0d684a758002/inventory.py#L99-L122

Reply:
I think I'll just improve it to only iterate through the expected number of pages. This was really just the quickie hack.

Reply:
Yeah, if you can change to `for` loop, that's probably better all around.

Reply:
Okay I rewrote this entire file for this ;)